TestFoundation/TestUUID.swift

```diff
@@ -0,0 +1,302 @@
+        XCTAssertNotEqual(uuidC, uuidA)
+        XCTAssertNotEqual(uuidG, uuidC)
+        XCTAssertNotEqual(uuidC, uuidG)
+    }
+    
+    func test_description() {
+        let uuidA = UUID(uuid: zero)
+        let uuidB = UUID(uuidString: zeroString)
+        let uuidC = UUID(uuid: test)
+        let uuidD = UUID(uuidString: testString)
+        let uuidE = UUID(uuid: hex)
+        
+        XCTAssertEqual(uuidA.description, zeroString)
+        XCTAssertEqual(uuidB?.description, zeroString)
+        XCTAssertEqual(uuidC.description, testString.uppercased())
+        XCTAssertEqual(uuidD?.description, testString.uppercased())
+        XCTAssertEqual(uuidA.description, uuidB?.description)
+        XCTAssertEqual(uuidC.description, uuidD?.description)
+        XCTAssertEqual(uuidE.description, validUUIDStrings[0])
+    }
+    
+    func test_debugDescription() {
+        let uuidA = UUID(uuid: zero)
+        let uuidB = UUID(uuidString: zeroString)
+        let uuidC = UUID(uuid: test)
+        let uuidD = UUID(uuidString: testString)
+        let uuidE = UUID(uuid: hex)
+        
+        XCTAssertEqual(uuidA.debugDescription, zeroString)
+        XCTAssertEqual(uuidB?.debugDescription, zeroString)
+        XCTAssertEqual(uuidC.debugDescription, testString.uppercased())
+        XCTAssertEqual(uuidD?.debugDescription, testString.uppercased())
+        XCTAssertEqual(uuidA.debugDescription, uuidB?.description)
+        XCTAssertEqual(uuidC.debugDescription, uuidD?.description)
+        XCTAssertEqual(uuidE.debugDescription, uuidE.description)
+    }
+    
+    func test_uuidString() {
+        let uuidA = UUID(uuid: zero)
+        let uuidB = UUID(uuidString: zeroString)
+        let uuidC = UUID(uuid: test)
+        let uuidD = UUID(uuidString: testString)
+        let uuidE = UUID(uuid: hex)
+        
+        XCTAssertEqual(uuidA.uuidString, zeroString)
+        XCTAssertEqual(uuidB?.uuidString, zeroString)
+        XCTAssertEqual(uuidC.uuidString, testString.uppercased())
+        XCTAssertEqual(uuidD?.uuidString, testString.uppercased())
+        XCTAssertEqual(uuidA.uuidString, uuidB?.description)
+        XCTAssertEqual(uuidC.uuidString, uuidD?.description)
+        XCTAssertEqual(uuidE.uuidString, validUUIDStrings[0])
+        XCTAssertEqual(uuidE.uuidString, uuidE.description)
+    }
+    
+    func test_customReflection() {
+        let uuid   = UUID()
+        let mirror = uuid.customMirror
+        let children = Array(mirror.children)
+        
+        XCTAssertNotNil(mirror)
+        XCTAssertEqual(mirror.displayStyle, .struct)
+        XCTAssertNil(mirror.superclassMirror)
+        XCTAssertEqual(children.count, 0)
+    }
+    
+    func test_bridgeToObjectiveC() {
+        let uuid    = UUID(uuid: test)
+        let bridged = uuid._bridgeToObjectiveC()
+        
+        XCTAssertNotNil(bridged)
+        XCTAssertTrue((bridged as Any) is NSUUID)
+        XCTAssertFalse((bridged as Any) is UUID)
```

Comment:
this line fails when running this test on Darwin (not in swift-corelibs-foundation) so it probably means there is a bug somewhere.